kohttp/src/main/kotlin/io/github/rybalkinsd/kohttp/client/ClientBuilderImpl.kt

```diff
@@ -141,5 +148,16 @@ internal class ClientBuilderImpl : ClientBuilder {
         set(value) { builder.pingInterval(value, TimeUnit.MILLISECONDS) }
         get() = throw UnsupportedOperationException()
 
-    fun build(): OkHttpClient = builder.build()
+    fun build(): OkHttpClient {
+        buildWithSslMaterialIfPresent()
+
+        return builder.build()
+    }
+
+    private fun buildWithSslMaterialIfPresent() {
+        if (internalSslSocketFactory != null && internalX509TrustManager != null) {
```

Comment:
Looks like `internalSslSocketFactory` and `internalX509TrustManager` are highly coupled.
IMO, it'll be great to introduce SslConfig bringing them together.

    var sslConfig(config: SslConfig)
        set(value) { builder.sslSocketFactory(...) }

It's a good question where it's better to introduce config class, will be happy to discuss options.

Reply:
Yes, they are indeed highly coupled. I had the feeling that this snippet of code didn't fit very well with the reaming code within the ClientBuilderImp as I first try to set them separately and then validate/add them to the builder during the build function.

I like the idea of a SslConfig. I created a data class within the base Config class and aligned the ClientBuilderImpl and ClientBuilder classes with the new property. I dropped the separate trustmanager and sslsocketfactory variables. Let me know what you think of the refactored solution and if additional adjustments are required.